usl_pipeline/cloud_functions/main_test.py

```diff
@@ -122,6 +126,89 @@ def test_build_feature_matrix(mock_storage_client, mock_firestore_client):
         {"elevation_min": 1, "elevation_max": 6},
     )
 
+@mock.patch.object(main.firestore, "Client", autospec=True)
+@mock.patch.object(main.storage, "Client", autospec=True)
+def test_build_feature_matrix_wrf(mock_storage_client, mock_firestore_client):
+    # Build a netcdf file
+    with netCDF4.Dataset('met_em_test.nc', mode='w', format='NETCDF4') as ncfile:
+        ncfile.createDimension('Time', 1)
+        ncfile.createDimension('west_east', 3)
+        ncfile.createDimension('south_north', 3)
+
+        time = ncfile.createVariable('time','f8',('Time',))
+        lat = ncfile.createVariable('lat', 'float32', ('west_east',))
+        lon = ncfile.createVariable('lon', 'float32', ('south_north',))
+        sno_alb = ncfile.createVariable('SNOALB','float32',('Time', 'south_north', 'west_east'))
+        
+        time[:] = 100
+        lat[:] = [200, 200, 200]
+        lon[:] = [300, 300, 300]
+        sno_alb[:] = numpy.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+    # Check the dataset written to the local ncfile
+    with xr.open_dataset("met_em_test.nc") as ds:
+        print("Checking contents of saved ncfile")
+        print(ds)
+
+    # Place the ncfile bytes into an archive.
```

Comment:
@waltaskew @rsutormin - Could you help take a look at how the unit test is set up? Currently, the unit test is failing when the `main.py:line306`. The netcdf file that was created in the test for some reason can't be loaded as an xarray dataset (error similar to: https://stackoverflow.com/questions/74453978/xarray-error-valueerror-did-not-find-a-match-in-any-of-xarrays-currently-ins), which points to either an incompatible netcdf file format or the file does not exist.

I manually tested the new code with an existing wps file we have on hand and was able to verify that the code worked e2e (reading archive -> uploading npy file). So not sure what i'm doing incorrectly in the unit test here.

Thanks!

Reply:
I'd wager it's because you aren't setting the size on the TarInfo object. Elsewhere in the file, we have
```
        tf = tarfile.TarInfo("elevation.tif")
        tf.size = len(tiff_bytes)
        tar.addfile(tf, io.BytesIO(tiff_bytes))
```
so without the size, I'd guess you're adding a size zero entry.